source/Halibut/Transport/ConnectionManager.cs

```diff
@@ -47,6 +45,30 @@ public async Task<IConnection> AcquireConnectionAsync(ExchangeProtocolBuilder ex
                 return openableConnection;
             }
         }
+        
+        async Task<IConnection> TryGetExistingConnectionAsync(ExchangeProtocolBuilder exchangeProtocolBuilder, IConnectionFactory connectionFactory, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken)
+        {
+            await Task.CompletedTask;
+            lock (activeConnections)
```

Comment:
I don't think we need this to be async, since no IO should ever occur within this lock. That is the reason the complex Tuple logic exists.